### ISSUE TITLE
Files with extremely long filenames do not get packaged

### ISSUE THREAD

**Issue by abenson** [void-linux/xbps#2]
It looks like files with paths over 127 characters will not get packaged. 

I created two directory structures, 1 with 127 characters, another with 128. The 127 character path gets added, the 128 character does not.

~~~
greenleaf% a=$(echo a{1..33} | sed 's| |/|g')
greenleaf% mkdir -p $a
greenleaf% touch $a/aaa
greenleaf% echo $a/aaa | wc -c
127
greenleaf% b=$(echo b{1..33} | sed 's| |/|g')
greenleaf% mkdir -p $b
greenleaf% touch $b/bbbb
greenleaf% echo $b/bbbb | wc -c
128
greenleaf% find . -type f
./a1/a2/a3/a4/a5/a6/a7/a8/a9/a10/a11/a12/a13/a14/a15/a16/a17/a18/a19/a20/a21/a22/a23/a24/a25/a26/a27/a28/a29/a30/a31/a32/a33/aaa
./b1/b2/b3/b4/b5/b6/b7/b8/b9/b10/b11/b12/b13/b14/b15/b16/b17/b18/b19/b20/b21/b22/b23/b24/b25/b26/b27/b28/b29/b30/b31/b32/b33/bbbb
greenleaf% cd ..
greenleaf% xbps-create -A noarch -n test-1.0_1 -s "testing filename length" testdest 
test-1.0_1: adding `/a1/a2/a3/a4/a5/a6/a7/a8/a9/a10/a11/a12/a13/a14/a15/a16/a17/a18/a19/a20/a21/a22/a23/a24/a25/a26/a27/a28/a29/a30/a31/a32/a33/a' ...
test-1.0_1: binary package created successfully (test-1.0_1.noarch.xbps)
greenleaf% tar Jtvf test-1.0_1.noarch.xbps
-rw-r--r-- root/root       496 1969-12-31 18:00 ./props.plist
-rw-r--r-- root/root     11652 1969-12-31 18:00 ./files.plist
-rw-r--r-- 0/0               0 2018-06-27 07:59 ./a1/a2/a3/a4/a5/a6/a7/a8/a9/a10/a11/a12/a13/a14/a15/a16/a17/a18/a19/a20/a21/a22/a23/a24/a25/a26/a27/a28/a29/a30/a31/a32/a33/a
~~~

**Comment by alexjarosch** [comment]
I can confirm that the atom package since version 1.26.1 is a victim of this behaviour and looses its GUI due to removal of files and directories in /usr/share/atom after installing the package with xbps

**Comment by abenson** [comment]
It seems to be an issue with portableproplib. It silently refuses keys over 128 characters long.

~~~
   else if (size <= PDK_SIZE_128)
      pdk = _PROP_POOL_GET(_prop_dictionary_keysym128_pool);
   else
      pdk = NULL; /* key too long */

(./lib/portableproplib/prop_dictionary.c:312)
~~~

`xbps_create` uses the filepath as the key.

~~~
   fileinfo = xbps_dictionary_create();
   /* XXX: fileinfo contains the sanatized path, whereas xe contains the
    * unsanatized path!
    *
    * when handing the files over, do not use the dictionary directly. Instead
    * use the keysym, as this value has the unsanatized path.
    */
   xbps_dictionary_set_cstring(fileinfo, "file", filep);
   xbps_dictionary_set(all_filesd, fpath, fileinfo);

(bin/xbps-create/main.c:322)
~~~

**Comment by Gottox** [comment]
Please check 3ef932de2f86b239bc7b0d5d3b8784b3c2ebdb74 if this solves the issue.

**Comment by alexjarosch** [comment]
I can confirm that the above test of @abenson as well as packaging atom with the new xbps (https://github.com/void-linux/xbps/commit/3ef932de2f86b239bc7b0d5d3b8784b3c2ebdb74) works perfectly.
I used a fresh void install to test, cloned void-packages and after a binary bootstrap I replaced the existing xbps-create within masterdir/usr/bin with a freshly compiled version of https://github.com/void-linux/xbps/commit/3ef932de2f86b239bc7b0d5d3b8784b3c2ebdb74.
Guess that solves the issue, great work @Gottox

**Comment by solbjorn** [comment]
@alexjarosch, perfect! Thanks for testing.
@Gottox, are there any plans to release a new stable with this critical fix (0.52.1 or smth like) or it's better to simply locally replace xbps-create in masterdir for now? Anyway, thanks for fixing out.
